diff -uNr qemu-5.1.0/configure qemu-5.1.0.mod/configure
--- qemu-5.1.0/configure	2020-08-11 22:17:15.000000000 +0300
+++ qemu-5.1.0.mod/configure	2020-09-05 16:56:55.236312855 +0300
@@ -131,7 +131,7 @@
 compile_prog() {
   local_cflags="$1"
   local_ldflags="$2"
-  do_cc $QEMU_CFLAGS $local_cflags -o $TMPE $TMPC $QEMU_LDFLAGS $local_ldflags
+  do_cc $CFLAGS $CPPFLAGS $QEMU_CFLAGS $local_cflags -o $TMPE $TMPC $LDFLAGS $QEMU_LDFLAGS $local_ldflags
 }
 
 # symbolically link $1 to $2.  Portable version of "ln -sf".
@@ -2550,8 +2550,8 @@
 #include <snappy-c.h>
 int main(void) { snappy_max_compressed_length(4096); return 0; }
 EOF
-    if compile_prog "" "-lsnappy" ; then
-        libs_softmmu="$libs_softmmu -lsnappy"
+    if compile_prog "" "-lsnappy -lc++_static -lc++abi" ; then
+        libs_softmmu="$libs_softmmu -lsnappy -lc++_static -lc++abi"
         snappy="yes"
     else
         if test "$snappy" = "yes"; then
@@ -3059,6 +3059,8 @@
 fi
 
 has_libgcrypt() {
+	return 0
+
     if ! has "libgcrypt-config"
     then
 	return 1
@@ -3130,10 +3132,7 @@
         # Debian has removed -lgpg-error from libgcrypt-config
         # as it "spreads unnecessary dependencies" which in
         # turn breaks static builds...
-        if test "$static" = "yes"
-        then
-            gcrypt_libs="$gcrypt_libs -lgpg-error"
-        fi
+        gcrypt_libs="$gcrypt_libs -lgpg-error"
 
         # Link test to make sure the given libraries work (e.g for static).
         write_c_skeleton
@@ -3485,6 +3484,7 @@
 # VNC SASL detection
 if test "$vnc" = "yes" && test "$vnc_sasl" != "no" ; then
   cat > $TMPC <<EOF
+#include <sys/types.h>
 #include <sasl/sasl.h>
 #include <stdio.h>
 int main(void) { sasl_server_init(NULL, "qemu"); return 0; }
@@ -3543,10 +3543,10 @@
 EOF
   if $pkg_config libpng --exists; then
     vnc_png_cflags=$($pkg_config libpng --cflags)
-    vnc_png_libs=$($pkg_config libpng --libs)
+    vnc_png_libs="$($pkg_config libpng --libs) -lm"
   else
     vnc_png_cflags=""
-    vnc_png_libs="-lpng"
+    vnc_png_libs="-lpng -lm"
   fi
   if compile_prog "$vnc_png_cflags" "$vnc_png_libs" ; then
     vnc_png=yes
@@ -3909,7 +3909,7 @@
 int main(void) { curl_easy_init(); curl_multi_setopt(0, 0, 0); return 0; }
 EOF
   curl_cflags=$($curlconfig --cflags 2>/dev/null)
-  curl_libs=$($curlconfig --libs 2>/dev/null)
+  curl_libs="$($curlconfig --libs 2>/dev/null) -lnghttp2 -lssl -lcrypto -lz -ldl"
   if compile_prog "$curl_cflags" "$curl_libs" ; then
     curl=yes
   else
@@ -3942,7 +3942,7 @@
 for i in $glib_modules; do
     if $pkg_config --atleast-version=$glib_req_ver $i; then
         glib_cflags=$($pkg_config --cflags $i)
-        glib_libs=$($pkg_config --libs $i)
+        glib_libs="$($pkg_config --libs $i) -landroid-support"
         QEMU_CFLAGS="$glib_cflags $QEMU_CFLAGS"
         LIBS="$glib_libs $LIBS"
         libs_qga="$glib_libs $libs_qga"
@@ -3954,7 +3954,7 @@
 if $pkg_config --atleast-version=$glib_req_ver gio-2.0; then
     gio=yes
     gio_cflags=$($pkg_config --cflags gio-2.0)
-    gio_libs=$($pkg_config --libs gio-2.0)
+    gio_libs="$($pkg_config --libs gio-2.0) -lgmodule-2.0 -lffi -landroid-support"
     gdbus_codegen=$($pkg_config --variable=gdbus_codegen gio-2.0)
     if [ ! -x "$gdbus_codegen" ]; then
         gdbus_codegen=
@@ -4070,7 +4070,7 @@
   pixman_libs=
 elif $pkg_config --atleast-version=0.21.8 pixman-1 > /dev/null 2>&1; then
   pixman_cflags=$($pkg_config --cflags pixman-1)
-  pixman_libs=$($pkg_config --libs pixman-1)
+  pixman_libs="$($pkg_config --libs pixman-1) -landroid-cpufeatures"
 else
   error_exit "pixman >= 0.21.8 not present." \
       "Please install the pixman devel package."
@@ -4756,6 +4756,7 @@
 if compile_prog "" "" ; then
   signalfd=yes
 fi
+signalfd=no
 
 # check if optreset global is declared by <getopt.h>
 optreset="no"
@@ -6525,9 +6526,9 @@
   QEMU_CFLAGS="-fprofile-arcs -ftest-coverage -g $QEMU_CFLAGS"
   QEMU_LDFLAGS="-fprofile-arcs -ftest-coverage $QEMU_LDFLAGS"
 elif test "$fortify_source" = "yes" ; then
-  CFLAGS="-O2 -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 $CFLAGS"
+  CFLAGS="-U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 $CFLAGS"
 elif test "$debug" = "no"; then
-  CFLAGS="-O2 $CFLAGS"
+  CFLAGS="$CFLAGS"
 fi
 
 if test "$have_asan" = "yes"; then
@@ -6631,6 +6632,7 @@
 if [ "$eventfd" = "yes" ]; then
   ivshmem=yes
 fi
+ivshmem=no
 
 tools=""
 if test "$want_tools" = "yes" ; then
@@ -7291,9 +7293,9 @@
 if test "$posix_fallocate" = "yes" ; then
   echo "CONFIG_POSIX_FALLOCATE=y" >> $config_host_mak
 fi
-if test "$sync_file_range" = "yes" ; then
-  echo "CONFIG_SYNC_FILE_RANGE=y" >> $config_host_mak
-fi
+#if test "$sync_file_range" = "yes" ; then
+#  echo "CONFIG_SYNC_FILE_RANGE=y" >> $config_host_mak
+#fi
 if test "$fiemap" = "yes" ; then
   echo "CONFIG_FIEMAP=y" >> $config_host_mak
 fi
